src/source/Ice/Network.c

```diff
@@ -11,15 +11,77 @@ STATUS getLocalhostIpAddresses(PKvsIpAddress destIpList, PUINT32 pDestIpListLen,
     UINT32 ipCount = 0, destIpListLen;
     BOOL filterSet = TRUE;
 
+#ifdef _WIN32
+    DWORD rv, size;
+    PIP_ADAPTER_ADDRESSES adapter_addresses, aa = NULL;
+    PIP_ADAPTER_UNICAST_ADDRESS ua;
+#else
     struct ifaddrs *ifaddr = NULL, *ifa = NULL;
+#endif
     struct sockaddr_in *pIpv4Addr = NULL;
     struct sockaddr_in6 *pIpv6Addr = NULL;
 
     CHK(destIpList != NULL && pDestIpListLen != NULL, STATUS_NULL_ARG);
     CHK(*pDestIpListLen != 0, STATUS_INVALID_ARG);
-    CHK(getifaddrs(&ifaddr) != -1, STATUS_GET_LOCAL_IP_ADDRESSES_FAILED);
 
     destIpListLen = *pDestIpListLen;
+#ifdef _WIN32
+    rv = GetAdaptersAddresses(AF_UNSPEC, GAA_FLAG_INCLUDE_PREFIX, NULL, NULL, &size);
+    CHK(rv == ERROR_BUFFER_OVERFLOW, STATUS_GET_LOCAL_IP_ADDRESSES_FAILED);
+
+    adapter_addresses = (PIP_ADAPTER_ADDRESSES)malloc(size);
+
+    rv = GetAdaptersAddresses(AF_UNSPEC, GAA_FLAG_INCLUDE_PREFIX, NULL, adapter_addresses, &size);
+    CHK(rv == ERROR_SUCCESS, STATUS_GET_LOCAL_IP_ADDRESSES_FAILED);
+    
+    for (aa = adapter_addresses; aa != NULL && ipCount < destIpListLen; aa = aa->Next) {
+
+        char ifa_name[BUFSIZ];
```

Comment:
CHAR ifaName. Importantly, move this to the top with the other local variables declarations. Some compilers will choke on this as CHK (goto) can jump over the definitions